src/main/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsDataSourceWriter.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zepben.evolve.database.sqlite.common.SchemaUtils
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.database.sqlite.extensions.configureBatch
+import com.zepben.evolve.metrics.IngestionJob
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import javax.sql.DataSource
+
+/**
+ * Class for writing metrics to an arbitrary datasource.
+ */
+class MetricsDataSourceWriter @JvmOverloads constructor(
+    private val dataSource: DataSource,
+    private val databaseTables: MetricsDatabaseTables = MetricsDatabaseTables()
+) {
+
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+    private val schemaUtils = SchemaUtils(databaseTables, logger)
+    private val versionTable = databaseTables.getTable<TableVersion>()
+
+    @Throws(IncompatibleVersionException::class)
+    fun save(job: IngestionJob): Boolean {
+        return dataSource.connection.use { connection ->
+            connection.configureBatch()
+            val localVersion = versionTable.supportedVersion
+            when (val remoteVersion = schemaUtils.getVersion(connection)) {
+                null -> schemaUtils.createSchema(connection) && populateTables(connection, job) && postSave(connection)
+                localVersion -> populateTables(connection, job) && postSave(connection)
+                else -> throw IncompatibleVersionException(localVersion, remoteVersion)
```

Comment:
todo: Is this deliberately designed differently to the other writers? The other writers return false for incompatible versions rather than throwing an exception, but this is not described in any docs or notes.

Can you please either add a note on why this has been implemented differently, or rework it to be consistent with the others.